assets/controller.yaml

```diff
@@ -41,17 +41,29 @@ spec:
                 secretKeyRef:
                   name: ebs-cloud-credentials
                   key: aws_access_key_id
+                  optional: true
             - name: AWS_SECRET_ACCESS_KEY
               valueFrom:
                 secretKeyRef:
                   name: ebs-cloud-credentials
                   key: aws_secret_access_key
+                  optional: true
+            - name: AWS_SDK_LOAD_CONFIG
+              value: 1
+            - name: AWS_CONFIG_FILE
```

Comment:
why `AWS_CONFIG_FILE` instead of `AWS_SHARED_CREDENTIALS_FILE`?

Reply:
i was just following:
https://github.com/aws/aws-sdk-go/blob/38c74caea1398949b67da14dfaa79cabe704a57f/aws/session/env_config.go#L78-L84

https://github.com/aws/aws-sdk-go/blob/38c74caea1398949b67da14dfaa79cabe704a57f/aws/session/session.go#L437-L442 shows that when we set `AWS_SDK_LOAD_CONFIG=1` and `AWS_CONFIG_FILE=<>` together to work.

i'm fine if want to converge to `AWS_SHARED_CREDENTIALS_FILE` ??

Reply:
I suppose it doesn't matter. Was just wondering if one was preferred over the other.

I remember some discussions early on about whether one of the env vars would work with STS and/or classic access/secret key data, and the other wouldn't work with both types of creds. All my validations were done with AWS_SHARED_CREDENTIALS_FILE (for the environments that supported it (golang and python)).

Reply:
Sorry my ignorance here, but why is it necessary to mount the secret as a volume and set `AWS_CONFIG_FILE` to it? Wouldn't `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` be enough? Or that's a requirement for using the STS credentials?

Reply:
AWS SDK supports initializing the credentials in multiple ways,

- currently in use are static creds set using these env AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY 
- but moving forward we want the SDK to load the credentials/config file (mounted from secret) and use that to configure the credentials.
  the config file supports all types of workflows for credentials, like static creds with access_key and secret key , basic assume_role or the new sts provider. Therefore using the config file allows customers in manual mode to provide credentials in AWS SDK standard supported ways including the STS.

Reply:
Ack, thank you for the detailed explanation.